src/repo/kv.rs

```diff
@@ -0,0 +1,387 @@
+use super::{Column, DataStore};
+use crate::error::Error;
+use crate::repo::{PinKind, PinMode, PinStore, References};
+use async_trait::async_trait;
+use cid::{self, Cid};
+use futures::stream::{StreamExt, TryStreamExt};
+use sled::{self, Config as DbConfig, Db, Mode as DbMode};
+use std::convert::Into;
+use std::path::PathBuf;
+use std::str::{self, FromStr};
+use std::sync::Mutex;
+use tracing_futures::Instrument;
+
+#[derive(Debug)]
+pub struct KvDataStore {
+    path: PathBuf,
+    db: Mutex<Option<Db>>,
```

Comment:
since the operations on `Db` take `self` by reference, it would probably be best to instead use something like an `OnceCell<Db>` here instead

Reply:
Is there a stable SyncOnceCell? I think those are still unstable.

Reply:
what about https://crates.io/crates/once_cell? it's post `1.0`

Reply:
Maybe it is better for not importing a new crates :)

Reply:
performing a `lock()` with every database operation (in addition to ones it's already using) is not ideal, though

Reply:
Got it, I have check once_cell, it is a good choice in this situation 👍 .